spec/requests/api/v2/providers/accredited_body_get_providers_spec.rb

```diff
@@ -0,0 +1,194 @@
+require "rails_helper"
+
+describe "AccreditedBody API v2", type: :request do
+  describe "GET /providers" do
+    let(:user) { create(:user, organisations: [organisation]) }
+    let(:organisation) { create(:organisation) }
+    let(:recruitment_cycle) { find_or_create :recruitment_cycle }
+    let(:payload) { { email: user.email } }
+    let(:token) do
+      JWT.encode payload,
+                 Settings.authentication.secret,
+                 Settings.authentication.algorithm
+    end
+    let(:credentials) do
+      ActionController::HttpAuthentication::Token.encode_credentials(token)
+    end
+
+    let!(:course) do
+      create(:course, provider: delivering_provider1, accrediting_provider: accredited_provider)
+    end
+    let!(:course2) do
+      create(:course, provider: delivering_provider2, accrediting_provider: accredited_provider)
+    end
+
+    let(:delivering_provider1) { create(:provider) }
+    let(:delivering_provider2) { create(:provider) }
+    let(:accredited_provider) {
+      create(:provider,
+             organisations: [organisation],
+             recruitment_cycle: recruitment_cycle)
+    }
+
+    let(:json_response) { JSON.parse(response.body) }
+    let(:request_path) { "/api/v2/recruitment_cycles/#{recruitment_cycle.year}/providers/#{accredited_provider.provider_code}/training_providers" }
+
+    def perform_request
+      get request_path, headers: { "HTTP_AUTHORIZATION" => credentials }
+    end
+
+    subject do
+      perform_request
+
+      response
+    end
+
+    context "when unauthorized" do
+      let(:payload) { { email: "foo@bar" } }
+
+      it { should have_http_status(:unauthorized) }
+    end
```

Comment:
Would it be worth using the shared example for unauthorised tests?

https://github.com/DFE-Digital/manage-courses-backend/blob/master/spec/support/shared_examples/unauth.rb

Reply:
It would as that uncovered a bug in the authorization. Absolute master stroke. Nice one 👍